### PR TITLE
Language synchronization

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR habilitates SyncLanguage to update a repo's existing language amount or create a repo new language.

### CODE REVIEW

File: app/services/sync/sync_language.rb [hunk 1]
```diff
@@ -6,26 +6,28 @@ def initialize(profile, repo, repo_language_data)
       self.repo_language_data = repo_language_data
     end
 
-    def create_language
-      new_language = Language.new
-      assign_attributes new_language
-      new_language.save
-      fetch_profile_and_repo new_language
+    def synchronize
+      self.language = repo.languages.find_by(name: repo_language_data[:name])
```
Comment (ImMPrada) [inline app/services/sync/sync_language.rb:10]: look for repo_language_data[:name] into repo's language

File: app/services/sync/fetch_repos.rb [hunk 2]
```diff
@@ -7,14 +7,35 @@ def initialize(profile)
 
     def call
       github_repos = Github::ReposConsumer.new(profile.nickname).call
-      github_repos.each do |github_repo|
-        repo = Sync::SyncRepo.new(profile, github_repo).create_repo
-        Sync::FetchLanguages.new(profile, repo).call
-      end
+      github_repos.each { |github_repo| syncronize_repo(github_repo) }
```
Comment (mapra99) [inline app/services/sync/fetch_repos.rb:10]:
```suggestion
      github_repos.each { |github_repo| synchronize_repo(github_repo) }
```

File: app/services/sync/fetch_repos.rb [hunk 3]
```diff
@@ -7,14 +7,35 @@ def initialize(profile)
...
     end
 
     private
 
-    attr_accessor :profile, :repos
+    attr_accessor :profile, :repos, :repo
+
+    def syncronize_repo(github_repo)
```
Comment (mapra99) [inline app/services/sync/fetch_repos.rb:17]:
```suggestion
    def synchronize_repo(github_repo)
```

File: app/services/sync/fetch_repos.rb [hunk 4]
```diff
@@ -7,14 +7,35 @@ def initialize(profile)
...
+    end
+
+    def create_repo_if_not_exists(github_repo)
+      return unless repo.nil?
+
+      self.repo = Sync::SyncRepo.new(profile, github_repo).create_repo
+      Sync::FetchLanguages.new(profile, repo).call
+    end
+
+    def syncronyze_repo(github_repo)
```
Comment (mapra99) [inline app/services/sync/fetch_repos.rb:31]: is this method repeated?

File: app/services/sync/fetch_repos.rb [hunk 5]
```diff
@@ -7,14 +7,35 @@ def initialize(profile)
...
+      this_repo_date = repo.git_date
```
Comment (mapra99) [inline app/services/sync/fetch_repos.rb:32]: feels very weird naming the vars with `this_*`. You already had stopped doing this a long ago on js
```suggestion
      repo_date = repo.git_date
```

File: app/services/sync/fetch_repos.rb (hunk 4 above)
Comment (ImMPrada) [reply app/services/sync/fetch_repos.rb]: please check again!

File: app/services/sync/fetch_profile.rb [hunk 6]
```diff
@@ -25,7 +25,7 @@ def create_profile_if_not_exists
       self.profile = Sync::SyncProfile.new(github_profile).create_profile
     end
 
-    def syncronyze_profile
+    def synchronize_profile
       this_profile_date = profile.git_date
```
Comment (mapra99) [inline app/services/sync/fetch_profile.rb:29]: `this_*`

**Review by mapra99 — APPROVED** [review]
go go go

**Pull request merged**